apps/dashboard/app/javascript/path_selector/path_selector_data_table.js

```diff
@@ -225,4 +227,24 @@ export class PathSelectorTable {
     data.files = filteredFiles;
     return data;
   }
+
+  filteredByFilename(file) {
+    if (this.filePattern !== "") {
```

Comment:
I can't tell if this expression covers the default. It seems like `this.filePattern` would be undefined here without a configuration, and `undefined !== ""` evaluates to true which to me seems unexpected.